### DISCUSSION TITLE
Is the USB hardware necessary?

### DISCUSSION THREAD

**Discussion by lxschwalb** [ctag-fh-kiel/ctag-tbd#54] [Q&A] [answered]
I recently discovered this project and I'm really excited about it. I will probably make myself 10 of them.

I'm wondering if I can replace the USB hardware with a header to plug in a usb to uart converter that can be reused for each module. This converter will likely be a CP2102N-A01-GQFN2 breakout board

So my question is, is USB used for anything other than uploading firmware?

**Comment by ctag-fh-kiel** [comment, marked as answer]
That should work, you'll need access to RX, TX, IO0 and EN lines to flash externally. USB is only needed for flashing. FW updates can be done through web access... Pg. 1 of the schematics illustrates the connections.

Note: TBD HW is CC BY-NC-SA 4.0 license https://creativecommons.org/licenses/by-nc-sa/4.0/

**Reply by lxschwalb** [reply]
Perfect, thanks
